Sources/StreamChat/Utils/SystemEnvironment+UserAgent.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+#if os(iOS)
+import UIKit
+#elseif os(macOS)
+import IOKit
+#endif
+
+extension SystemEnvironment {
+    static let userAgent: String? = {
+        guard let info = Bundle.main.infoDictionary,
+              let appName = (info["CFBundleDisplayName"] ?? info[kCFBundleIdentifierKey as String]) as? String,
+              let appVersion = info[kCFBundleVersionKey as String] as? String
+        else {
+            return nil
+        }
+
+        var model: String
+        var os: String
+
+        #if canImport(UIKit)
+
+        var scale = String(format: "%0.2f", UIScreen.main.scale)
+        model = UIDevice.current.model
+        os = UIDevice.current.systemVersion
+        return "\(appName)/\(appVersion) (\(deviceModelName); iOS \(os); Scale/\(scale))"
+        #else
+
+        model = getModelIdentifier() ?? ""
```

Comment:
You are assuming here that not being able to import UIKit means that getModelIdentifier is available (which is only available for macOS). This is true for now, but we should make those checks safer. Also, instead of empty strings, it would be better to add human readable fallbacks

Reply:
Done